### PR TITLE
Add documentation to ImplicitlyAnimatedWidgetState

### PR DESCRIPTION
**Pull request by tvolkert**
## Description

The documentation for `ImplicitlyAnimatedWidgetState ` was ambiguous in a few places.  This PR attempts to clarify those docs.  

This is a follow-on to #33370 based on review comments therein.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/implicit_animations.dart [hunk 1]
```diff
@@ -259,15 +257,36 @@ abstract class ImplicitlyAnimatedWidget extends StatefulWidget {
 ///
 /// This is the type of one of the arguments of [TweenVisitor], the signature
 /// used by [AnimatedWidgetBaseState.forEachTween].
+///
+/// Instances of this function are expected to take a value and return a tween
+/// beginning at that value.
+/// beginning at that value.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/implicit_animations.dart]: remove duplication?

File: packages/flutter/lib/src/widgets/implicit_animations.dart [hunk 2]
```diff
@@ -259,15 +257,36 @@ abstract class ImplicitlyAnimatedWidget extends StatefulWidget {
...
 typedef TweenConstructor<T> = Tween<T> Function(T targetValue);
 
-/// Signature for callbacks passed to [AnimatedWidgetBaseState.forEachTween].
+/// Signature for callbacks passed to [ImplicitlyAnimatedWidgetState.forEachTween].
+///
+/// [ImplicitlyAnimatedWidgetState.forEachTween] will be passed an instance of
+/// this visitor and is expected to call the visitor with the appropriate
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/implicit_animations.dart]: nit: maybe split this paragraph into two sentences to make i easier to read and comprehend?

File: packages/flutter/lib/src/widgets/implicit_animations.dart [hunk 3]
```diff
@@ -259,15 +257,36 @@ abstract class ImplicitlyAnimatedWidget extends StatefulWidget {
...
+/// arguments for each tween that the state manages -- and to set the tween
+/// value to the return value of this visitor.
+///
+/// The `tween` argument should contain the current tween value. This will
+/// initially be null when the state is first initialized.
+///
+/// The `targetValue` argument should contain the value toward which the state
+/// is animating. For instance, if the state was animating its widget's
+/// "opacity" value, then this argument should contain the widget's current
+/// opacity value.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/implicit_animations.dart]: Maybe unify the quotes around opacity? Either use them in both instances or don't?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/implicit_animations.dart [hunk 4]
```diff
@@ -423,6 +442,20 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
...
   /// updated within [didUpdateTweens], not within [forEachTween].
+  ///
+  /// This method will be called both:
+  ///
+  ///  1. After the tweens are _initially_ constructed (by
+  ///     the `constructor` argument to the [TweenVisitor] that's passed to
+  ///     [forEachTween]). In this case, the tweens are likely to contain only
+  ///     a [Tween.begin] value and not a [Tween.end].
+  ///
+  ///  2. When the state's [widget] is updated, and one or more of the tweens
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/implicit_animations.dart]: This is the confusingly simple bit. The animation's don't start immediately. Each time the didUpdateWidget method of the ImplicitlyAnimatedWidgetState subclass runs, they might start, per the condition you've specified. In other words, each time the ImplicitlyAnimatedWidget subclass is rebuilt, the animations might start.

**Review by HansMuller — APPROVED** [review]
LGTM

I think the forEachTween doc would also be worth clarifying. It would be good if the doc made it crystal clear, at the outset, that the subclass's override needs to apply the visitor callback to each of its tweens (not "animatable facet" even if that's somehow more correct).

File: packages/flutter/lib/src/widgets/implicit_animations.dart (hunk 4 above)
Comment (tvolkert) [reply packages/flutter/lib/src/widgets/implicit_animations.dart]: Yep.  I tried to add a lot more documentation to `forEachTween()` to make this clear.

**Pull request merged**